### PR TITLE
Magma: add version 2.4.0

### PR DESCRIPTION
**Pull request by luszczek**

### CODE REVIEW

File: var/spack/repos/builtin/packages/magma/package.py [hunk 1]
```diff
@@ -35,6 +35,7 @@ class Magma(CMakePackage):
     homepage = "http://icl.cs.utk.edu/magma/"
     url = "http://icl.cs.utk.edu/projectsfiles/magma/downloads/magma-2.2.0.tar.gz"
 
+    version('2.4.0', 'fe15b98a0c4fed3eec6dc7a0c36765c4')
```
[the remark this answers is not in the document]
Comment (luszczek) [reply var/spack/repos/builtin/packages/magma/package.py]: 4eb839b1295405fd29c8a6f5b4ed578476010bf976af46573f80d1169f1f9a4f

[the remark this answers is not in the document]
Comment (luszczek) [reply var/spack/repos/builtin/packages/magma/package.py]: I don't understand this comment at all.

Spack supports all hashes and none of them is preferred over others:

https://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=md5#checksums

In the past, this package had MD5 and it continues to have MD5.

I made a pull request as simple as possible and only added one short line.

This is as minimal of a change as I can make it so it's easy to review.

If you'd like to suggest a change of hashes then we can do it in a separate commit.

[the remark this answers is not in the document]
Comment (luszczek) [reply var/spack/repos/builtin/packages/magma/package.py]: I switched all the hashes to SHA256 so it's consistent and more secure going forward.

[the remark this answers is not in the document]
Comment (healther) [reply var/spack/repos/builtin/packages/magma/package.py]: we switched `spack checksum` over to sha256 some time ago, while md5 is technically not considered insecure for our purposes new contributions are encouraged to be sha256 (I'm not sure whether we handle this as a strict requirement thought)

Thanks for changing it

**Comment by scheibelp** [comment]
Thanks!

**Pull request merged**